mlflow/projects/utils.py

```diff
@@ -80,3 +99,150 @@ def _is_valid_branch_name(work_dir, version):
         except GitCommandError:
             return False
     return False
+
+
+def fetch_and_validate_project(uri, version, entry_point, parameters):
+    parameters = parameters or {}
+    work_dir = fetch_project(uri=uri, force_tempdir=False, version=version)
+    project = _project_spec.load_project(work_dir)
+    project.get_entry_point(entry_point)._validate_parameters(parameters)
+    return project, work_dir
+
+
+def fetch_project(uri, force_tempdir, version=None):
```

Comment:
```suggestion
def fetch_project(uri, version=None):
```

I think we can remove force_tempdir, as all usages pass `force_tempdir=False`